### PR TITLE
Run next/link codemod for Next.js 13 on examples

### PR DESCRIPTION
**Pull request by timneutkens**
- Run link codemod on examples
- Run prettier

Ensures the examples show `<Link>` usage correctly.

## Bug

- [ ] Related issues linked using `fixes #number`
- [ ] Integration tests added
- [ ] Errors have a helpful link attached, see `contributing.md`

## Feature

- [ ] Implements an existing feature request or RFC. Make sure the feature request has been accepted for implementation before opening a PR.
- [ ] Related issues linked using `fixes #number`
- [ ] Integration tests added
- [ ] Documentation added
- [ ] Telemetry added. In case of a feature if it's used or not.
- [ ] Errors have a helpful link attached, see `contributing.md`

## Documentation / Examples

- [ ] Make sure the linting passes by running `pnpm lint`
- [ ] The "examples guidelines" are followed from [our contributing doc](https://github.com/vercel/next.js/blob/canary/contributing/examples/adding-examples.md)

### CODE REVIEW

File: examples/api-routes-rest/pages/index.tsx [hunk 1]
```diff
@@ -14,7 +14,7 @@ export default function Index() {
     <ul>
       {data.map((user) => (
         <li key={user.id}>
-          <Link href="/user/[id]" as={`/user/${user.id}`}>
+          <Link href="/user/[id]" as={`/user/${user.id}`} legacyBehavior>
```
Comment (styfle) [inline examples/api-routes-rest/pages/index.tsx:17]: Does this one need to be legacy? Looks like the child is already correct

File: examples/auth0/components/header.tsx [hunk 2]
```diff
@@ -11,30 +11,30 @@ const Header = ({ user, loading }: HeaderProps) => {
       <nav>
         <ul>
           <li>
-            <Link href="/">
+            <Link href="/" legacyBehavior>
               <a>Home</a>
```
Comment (styfle) [inline examples/auth0/components/header.tsx:15]: Seems like a bug: i would expect this anchor to be removed

**Review by balazsorban44 — CHANGES_REQUESTED** [review]
There are some edge cases that seemingly haven't fully been considered by the codemod, so we might want to fix that first and add some tests. We could fix https://github.com/vercel/next.js/issues/41925 at the same time too

File: examples/auth0/components/header.tsx (hunk 2 above)
Comment (ishaqibrahimbot) [reply examples/auth0/components/header.tsx:15]: Hey @styfle!

I was just taking a look around this PR and decided to check this out. The transform not removing the anchor tags in this file and instead adding the legacyBehavior prop is intentional. This is because of the presence of a <style> tag with a jsx attribute in this component which makes removing the anchor tag a bit risky in case the anchor tag has some CSS applied to it inside the <style> tag. See screenshot below and refer to [this commit](https://github.com/ishaqibrahimbot/next.js/commit/20e5fed1cf9973df5d4ea31b2193e77d8ac6140f) by Tim.

Screen Shot 2022-10-29 at 11 35 53 PM

Do you think it would be wise to log a simple warning message in this case? So that anyone running the codemod is alerted about any file(s) that might be treated like this because of the presence of <style jsx>? I can submit a PR with the warning message added. Thanks!

Comment (styfle) [reply examples/auth0/components/header.tsx:15]: Do we know why a style tag would affect the link? Seems like emitted HTML should be the same since “Link” becomes “a” with the new behavior.

Comment (ishaqibrahimbot) [reply examples/auth0/components/header.tsx:15]: This is probably because of scope limitations, i.e. the <style jsx> affects any <a> tags inside the same parent component's scope, but not <a> tags that are returned from React components inside that same parent component.

See this example in the [styled-jsx docs](https://github.com/vercel/styled-jsx) for a better explanation:

Screen Shot 2022-10-30 at 5 17 03 PM

I suppose this is why the decision was taken to just add the legacyBehavior prop if a <style jsx> tag is found in a file.

**Review by styfle — APPROVED** [review]

**Comment by styfle** [comment]
@balazsorban44 @timneutkens Lets merge this? 

Seems like the issues it will fix are greater than the issues it introduces. 

For example, it should close

- https://github.com/vercel/next.js/pull/42114
- https://github.com/vercel/next.js/pull/42181
